kalite/topic_tools/content_recommendation.py

```diff
@@ -194,85 +194,265 @@ def get_exercise_prereqs(exercises):
     return prereqs
     
 
+
+
 ########################################## 'EXPLORE' LOGIC ################################################
 
 ###
-# Returns a list of exercises that the user has not explored yet.
+# Returns a list of subtopic ids that the user has not explored yet. 
+#
+# @param: user - the facility user model corresponding to the current user
 #
-# @param: subtopic_id: the subtopic id of the most recent subtopic that the user has engaged in (possibly 
-#         averaged).
 # @return: a list of exercise id's of the 'middle to farthest neighbors,' or less immediately relevant
 #           exercises based on topic tree structure.
 ###
-def get_explore_recommendations(subtopic_id):
-    data = generate_recommendation_data()[subtopic_id]['related_subtopics']
+def get_explore_recommendations(user):
+
+    ''' 
+        Logic: grab 3 random exercises from recent exercises, get their subtopic ids, then using
+        generate_recommendation_data(), get the elements at certain positions (nearest). 
+
+    '''
+
+    #user = ExerciseLog.objects.filter(id__lte=1)[4].user            #random person, can delete after   
+
+    data = generate_recommendation_data()                           #topic tree alg
+    exercise_parents_table = get_exercise_parents_lookup_table()    #for finding out subtopic ids
+    recent_exercises = get_most_recent_exercises(user)              #most recent ex
+
+    #simply getting a list of subtopics accessed by user
+    recent_subtopics = []
+    for ex in recent_exercises:
+        if not exercise_parents_table[ex]['subtopic_id'] in recent_subtopics:
+            recent_subtopics.append(exercise_parents_table[ex]['subtopic_id'])
+
+    #choose sample number, up to three
+    if len(recent_exercises) > 0:
+        sampleNum = 1 #must be at least 1
+
+        if len(recent_exercises) > 1:
+            sumpleNum = 2
+
+            if len(recent_exercises) > 2:
+                sampleNum = 3
+
+    else:
+        sampleNum = 0 #user has not attempted any
     
-    data = data[(len(data)/2):]                     #only look at middle to furthest neighbors
-   
-    ''' Need to add some more analysis to ensure that the user has not done these exercises yet'''
+    random_exercises = random.sample(recent_exercises, sampleNum)   #grab the valid/appropriate number of exs, up to 3
+    added = []                                                      #keep track of what has been added (below)
+    final = []                                                      #final recommendations to return
+    
+    for ex in random_exercises:
+        exercise_data = exercise_parents_table[ex]
+        subtopic_id = exercise_data['subtopic_id']                      #subtopic_id of current
+        related_subtopics = data[subtopic_id]['related_subtopics'][2:7] #get recommendations based on this, can tweak numbers!
```

Comment:
Is there an explanation into why these specific numbers are chosen? Can you also move this to module level variables?